addons/stock/models/product_strategy.py

```diff
@@ -110,7 +110,7 @@ def _get_putaway_location(self, product, quantity=0, package=None, qty_by_locati
                     return location_out
                 continue
 
-            child_locations = self.env['stock.location'].search([('id', 'child_of', location_out.id), ('usage', '=', 'internal')])
+            child_locations = location_out.child_internal_location_ids
```

Comment:
I am afraid of the prefetch here. The compute of `child_internal_location_ids` will be execute for all putaway_rule.location_out_id in self which is not lazy (opposite to the remaining code of the method).
Maybe a `location_out.with_prefetch().child_internal_location_ids` to cancel the prefetch (or maybe batch the `_compute_child_internal_location_ids`). What do you think ?

Reply:
I'm in favor for batch computation, though it's not that simple (we need to mix all children first and then sort them by ancestors). I've updated the PR, take a look please